src/EFCore.PG/Query/ExpressionTranslators/Internal/NpgsqlNetworkTranslator.cs

```diff
@@ -46,11 +46,21 @@ public class NpgsqlNetworkTranslator : IMethodCallTranslator
         [CanBeNull]
         public Expression Translate(MethodCallExpression expression)
         {
-            if (expression.Method.DeclaringType != typeof(NpgsqlNetworkExtensions))
+            var declaringType = expression.Method.DeclaringType;
+
+            if (declaringType != typeof(NpgsqlNetworkExtensions) &&
+                declaringType != typeof(IPAddress) &&
+                declaringType != typeof(PhysicalAddress))
                 return null;
 
             switch (expression.Method.Name)
             {
+            case nameof(IPAddress.Parse) when declaringType == typeof(IPAddress):
+                return new ExplicitCastExpression(expression.Arguments[0], typeof(IPAddress));
```

Comment:
Continuing the OCD mood from above, if the expression is a constant we may want to use the literal PostgreSQL syntax: `IPADDR '127.0.0.1'` rater than the cast syntax `CAST('127.0.0.1' AS IPADDR)`. The former is cleaner SQL and clearly indicates the constant-ness in the expression (and therefore cannot also be parameterized). You can use SqlFragmentExpression to easily generate it.

However, this is really a nit and we don't have to do it.

Reply:
We don't actually have a chance to do this, per https://github.com/npgsql/Npgsql.EntityFrameworkCore.PostgreSQL/issues/534#issuecomment-407144319. Though, I like the idea about `SqlFragmentExpression`.